data_prep/pure_graph_dataset.py

```diff
@@ -1,24 +1,58 @@
+import torch
+
 from data_prep.graph_dataset import GraphDataset
 
 
 class PureGraphDataset(GraphDataset):
     """
     Text Dataset used by the pure graph model.
     """
-    def __init__(self, *args, **kwargs):
-        # class super init
-        # append features to data object
-        raise NotImplementedError
+    def __init__(self, corpus, device):
+        super().__init__(corpus, device)
+        print('Generating feature matrix')
+        self._data.x = self._generate_features(self._tokens, self._raw_texts)
```

Comment:
Why passing the self._raw_texts here? can just be accessed within the generate features as the `self` object is anyways present in that method